### ISSUE TITLE
Formatting problem with characters like ☺☻♦

### ISSUE THREAD

**Issue by Suraj-Yadav** [microsoft/vscode-cpptools#176]
Consider the following code

``` cpp
#include <iostream>
int main() {
std::cout << "Hello World\n";
return 0;
}
```

Adding ♦ in the string and formatting the code, I get

``` cpp
#include <iostream>
int main() {
    std::cout << "Hello ♦ World\n";
    return 0;
}
```

Removing the ♦ and formatting it again, I get

``` cpp
#include <iostream>
int main() {
    std::cout << "Hello�� World\n";
    return 0;
}
```

After this each format action doubles the number of �.

Using clang-format directly on the file does not have this problem.

Version: 1.5.0-insider
Commit: 31f1acb8c82b8a19256d5d56127d3f81edce3745
OS: Windows 10

**Comment by sean-mcmanus** [comment]
Thanks for reporting this. The bug still repros in 0.9. The exact behavior is somewhat random. I should have a fix in the next update.

**Comment by Suraj-Yadav** [comment]
With the 0.9.0 update, the problem appears to have worsened. Now a code like

``` cpp
#include <iostream>
#include <unordered_map>

int main()
{  
    std::unordered_map<int,char> example = {{1,'a'},{2,'b'}};

    auto search = example.find(2);
    if(search != example.end()) {
        std::cout << "Found " << search->first << " " << search->second << '\n';
    }
    else {
        std::cout << "Not found\n";
    }
}
```

after formatting becomes

``` cpp
include <iosa{> clude <unordered_map>

int main()
{  
     s td: :unorderd_   int,char> example = {{1,'a'},{2,    }; 

    auto search = exampl        );
    if(search != example.end()) {
        std::cout << "Found " << se ->f  << " "         h->second << '\n';
    }
    se {
        std::cout << "Not found\n";
    }
}
```

Again using clang-format directly works perfectly.

cpptools Version 0.9.0
clang-format version 3.8.1 (branches/release_38)
Version 1.5.0-insider
Commit 1de4d832ccb061c13cf70eaf0221fbaa86abfbdd
Date 2016-08-31T06:10:03.069Z

**Comment by sean-mcmanus** [comment]
0.9 not was not the "next update". We didn't know about this problem until after 0.9 shipped. It should be fixed in a future update.

**Comment by sean-mcmanus** [comment]
I fixed it, but it won't make it into the upcoming 0.9.1 release (should be in the next one).

**Comment by sean-mcmanus** [comment]
Fixed in 0.9.2.
